pkg/querier/matrix.go

```diff
@@ -16,7 +16,8 @@ import (
 	"github.com/grafana/mimir/pkg/util/modelutil"
 )
 
-func mergeChunks(chunks []chunk.Chunk, from, through model.Time) chunkenc.Iterator {
+// TODO: reuse iterators.
```

Comment:
[nit] Not sure this TODO adds much value. If you're going to do it in a follow up PR, then there's no need for this TODO. If you're not going to do it in a follow up PR, then this TODO won't help much to find stuff to work on (an issue may be better).

Reply:
I do not plan to implement iterator re-use for them. I will remove the TODO.